shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -23,31 +24,88 @@ function _getLocalStorageObject(): Storage {
+ * @param storageType - Type of storage
+ * @returns {boolean} Returns whether storage object is safely accessible
+ */
+function _canSafelyAccessStorage(storageType: StorageType): boolean {
+    const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
+    let result = true;
+    
+    try {
+        // First, check if window exists and get the global object once
+        const gbl: any = getGlobal();
+        if (isNullOrUndefined(gbl)) {
+            result = false;
+        } else {
+            // Try to indirectly check if the property exists and is accessible
+            // This avoids direct property access that might throw in Safari with "Block All Cookies" enabled
+            
+            // Some browsers throw when accessing the property descriptors with getOwnPropertyDescriptor
+            // Others throw when directly accessing the storage objects
+            // This approach tries both methods safely
+            try {
+                // Method 1: Try using property descriptor - safer in Safari with cookies blocked
+                const descriptor = objGetOwnPropertyDescriptor(gbl, storageTypeName);
+                if (!descriptor || !descriptor.get) {
+                    result = false;
+                }
+            } catch (e) {
+                // If the above fails, attempt a direct access inside a try-catch
+                try {
+                    const storage = gbl[storageTypeName];
+                    if (!storage) {
+                        result = false;
+                    }
+                } catch (e) {
+                    // If both approaches fail, storage cannot be safely accessed
+                    result = false;
+                }
+            }
+        }
+    } catch (e) {
+        result = false;
+    }
+    
+    return result;
+}
+
 /**
  * Tests storage object (localStorage or sessionStorage) to verify that it is usable
  * More details here: https://mathiasbynens.be/notes/localstorage-pattern
  * @param storageType - Type of storage
  * @returns {Storage} Returns storage object verified that it is usable
  */
 function _getVerifiedStorageObject(storageType: StorageType): Storage {
+    let result = null;
+    
     try {
-        if (isNullOrUndefined(getGlobal())) {
-            return null;
-        }
-        let uid = (new Date).toString();
-        let storage: Storage = getGlobalInst(storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage");
-        let name:string = _storagePrefix + uid;
-        storage.setItem(name, uid);
-        let fail = storage.getItem(name) !== uid;
-        storage.removeItem(name);
-        if (!fail) {
-            return storage;
+        // First check if we can safely access the storage object
+        if (_canSafelyAccessStorage(storageType)) {
+            const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
```

Comment:
pass the resolved string into the _canSafelyAccessStorage to avoid the duplication of the strings